### ISSUE TITLE
Add more characters to valid name spec

### ISSUE THREAD

**Issue by heyitsmeuralex** [decent-chat/decent#183]
>Names may consist only of alphanumeric characters, underscores (_), and dashes (-).

Could we consider adding some more characters to this? For example, a dot/full-stop/period (`.`)?

@towerofnix: opinions pls

**Comment by towerofnix** [comment]
Yeah, these would be cool. Min/max length is pretty easy to implement as a server setting (a little more effort to make it settable through the client).

**Comment by towerofnix** [comment]
> Would be really cool to see a server config option to allow non-unique channel names. This'd be greatly useful for creating Decent-backed games, for example a per-user unique #inventory channel.

OHH YES,

**Comment by PullJosh** [comment]
Could you elaborate on the non-unique channel names thing? I'm confused. :stuck_out_tongue:

**Comment by towerofnix** [comment]
Basically, let multiple channels have the same name. (Internally, these channels would still have different IDs, of course.) This is handy for anybody who wants to create a bot user which runs a game. I guess you never saw dientendre/_v, two Discord-based games we were working on a while ago? (Also, if we ever do channel groups or whatever, handy then too.)

**Comment by heyitsmeuralex** [comment]
Would be nice if emotes supported any character except `:` - I'll do that now. At the moment, you can't call an emote `:+1:`!

**Comment by heyitsmeuralex** [comment]
> Would be really cool to see a server config option to allow non-unique channel names. This'd be greatly useful for creating Decent-backed games, for example a per-user unique #inventory channel.

Iteration of this: a `{ requireUnique: true }` default on `POST /api/channels`.
